.github/workflows/test.yml

```diff
@@ -0,0 +1,60 @@
+name: "test"
+
+on:
+  pull_request:
+    branches:
+      - main
+      - dev
+
+jobs:
+  test-package:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        dir: 
+          - 'file-upload'
+          - 'interceptor'
+          - 'monitoring'
+          - 'temporal' 
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4.1.7
+
+      - name: Use Node.js 
+        uses: actions/setup-node@v4
+
+      - name: Install Dependencies for packages/common
+        run: npm ci
+        working-directory: packages/common
+
+      - name: Run tests
+        run: yarn run test
```

Comment:
use npm only to run tests, don't use different package manager for installation and different for running tests.